Dockerfile

```diff
@@ -1,6 +1,26 @@
+#
+# BUILDER
+#
+FROM docker.io/library/golang:1.21.1 as builder
+
+WORKDIR /buildroot
+COPY ./ ./
+
+RUN go build -o controller-manager cmd/controller-manager/*.go
```

Comment:
```suggestion
# Copy the Go Modules manifests.
COPY go.mod go.mod
COPY go.sum go.sum

# Cache deps before building and copying source, so that we don't need to re-download
# as much and so theat source changes don't invalidate our downloaded layer.
RUN go mod download

# Copy the source.
COPY cmd/ cmd/
COPY pkg/ pkg/

# Disable CGO.
ENV CGO_ENABLED=0

# Build the image.
RUN go build -o build/controller-manager cmd/controller-manager/*.go
```

This is not necessary for CI but will allow caching the layers on manual builds.